clang/include/clang/Sema/Attr.h

```diff
@@ -32,5 +36,87 @@ inline bool isFunctionOrMethodOrBlockForAttrSubject(const Decl *D) {
   return isFuncOrMethodForAttrSubject(D) || llvm::isa<BlockDecl>(D);
 }
 
+/// Return true if the given decl has a declarator that should have
+/// been processed by Sema::GetTypeForDeclarator.
+inline bool hasDeclarator(const Decl *D) {
+  // In some sense, TypedefDecl really *ought* to be a DeclaratorDecl.
+  return isa<DeclaratorDecl>(D) || isa<BlockDecl>(D) ||
+         isa<TypedefNameDecl>(D) || isa<ObjCPropertyDecl>(D);
```

Comment:
```suggestion
  return isa<DeclaratorDec, BlockDecl, TypedefNameDecl, ObjCPropertyDecl>(D);
```

Reply:
We agreed to leave this out of scope of this PR, even though it makes perfect sense and should be done.